core/amber/src/main/scala/edu/uci/ics/texera/web/resource/aiassistant/AiAssistantManager.scala

```diff
@@ -1,16 +1,34 @@
 package edu.uci.ics.texera.web.resource.aiassistant
+
 import edu.uci.ics.amber.engine.common.AmberConfig
 import java.net.{HttpURLConnection, URL}
+import com.typesafe.config.Config
 
 object AiAssistantManager {
-  private val aiAssistantConfig = AmberConfig.aiAssistantConfig.getOrElse(
-    throw new Exception("ai-assistant-server configuration is missing in application.conf")
-  )
-  val assistantType: String = aiAssistantConfig.getString("assistant")
-  // The accountKey is the OpenAI authentication key used to authenticate API requests and obtain responses from the OpenAI service.
+  // Optionally retrieve the configuration
+  private val aiAssistantConfigOpt: Option[Config] = AmberConfig.aiAssistantConfig
+
+  // Public variables, accessible from outside the object
+  var accountKey: String = _
+  var sharedUrl: String = _
+
+  // Initialize accountKey and sharedUrl if the configuration is present
+  aiAssistantConfigOpt.foreach { aiAssistantConfig =>
```

Comment:
why using `foreach` here? This `aiAssistantConfigOpt` is not an array or list right?

Reply:
Option[X] is a collection of single element of type X. so `foreach` (and also other collection-based methods) can work here.